pysatModels/tests/test_utils_testing.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (C) 2022, pysat development team
+# Full license can be found in License.md
+# -----------------------------------------------------------------------------
+"""Unit tests for `pysatModels.utils.testing`."""
+
+import numpy as np
+from packaging import version as pack_version
+import pytest
+
+import pysat
+from pysat.instruments import pysat_testmodel
+
+from pysatModels.utils.testing import compare_mod_name_coordinates
+
+
+class TestUtilsCompareModName(object):
+    """Unit tests for `compare_mod_name_coordinates`."""
+
+    def setup(self):
+        """Set up the unit test environment for each method."""
+
+        self.model = pysat.Instrument(inst_module=pysat_testmodel, tag='')
+
+        # Load the data in the instruments
+        load_kwargs = {'date': pysat_testmodel._test_dates['']['']}
+        if(pack_version.Version(pysat.__version__)
+           > pack_version.Version('3.0.1')):
+            load_kwargs['use_header'] = True
+
+        self.model.load(**load_kwargs)
+
+        return
+
+    def teardown(self):
+        """Clean up the unit test environment after each method."""
+
+        del self.model
+        return
+
+    def test_failure_non_timelike_time(self):
+        """Test the failure when first dimension not timelike."""
+
+        # Replace time values with integers
+        self.model['time'] = np.arange(0, len(self.model['time']))
+
+        # Run, raising a value error
+        with pytest.raises(ValueError) as err:
+            compare_mod_name_coordinates(self.model['dummy1'],
+                                         ['latitude', 'longitude'])
+
+        assert str(err).find('does not appear to be a time dimension') >= 0
+
+        return
+
+    def test_failure_too_many(self):
+        """Test the failure when `mod_name` too long."""
+
+        # Run, raising a value error
+        with pytest.raises(ValueError) as err:
+            compare_mod_name_coordinates(self.model['dummy1'],
+                                         ['latitude', 'longitude', 'altitude'])
+
+        assert str(err).find('Provided too many ') >= 0
+
+        return
```

Comment:
These and probably more can be combined using parametrize.

Reply:
The generalized test was right below. It was even checking for time. Funny how I could look at those when putting it together but not see it.

Reply:
Thanks for catching the test issue. Have a good one!